sim/Makefile

```diff
@@ -50,21 +54,27 @@ BLOOP_NAILGUN_PORT ?= 8212
 SCALA_BUILDTOOL_DEPS ?= build.sbt
 
 ifdef ENABLE_BLOOP
-override SCALA_BUILDTOOL_DEPS += $(BLOOP_CONFIG_DIR)/TIMESTAMP
-# Two notes about the bloop invocation:
-# 1) the sed removes a leading {file:<path>} that sometimes needs to be
-#    provided to SBT when a project but not for bloop.
-# 2) Generally, one could could pass '--' to indicate all remaining arguments are
-# 	 destined for the scala Main, however a bug in Bloop's argument parsing causes the
-#    --nailgun-port argument to be lost in this case. Workaround this by prefixing
-#    every main-destined argument with "--args"
-define run_scala_main
-	cd $(base_dir) && bloop --nailgun-port $(BLOOP_NAILGUN_PORT) run $(shell echo $(1) | sed 's/{.*}//') --main $(2)  $(addprefix --args ,$3)
-endef
+  override SCALA_BUILDTOOL_DEPS += $(BLOOP_CONFIG_DIR)/TIMESTAMP
+  # Two notes about the bloop invocation:
+  # 1) the sed removes a leading {file:<path>} that sometimes needs to be
+  #    provided to SBT when a project but not for bloop.
+  # 2) Generally, one could could pass '--' to indicate all remaining arguments are
+  # 	 destined for the scala Main, however a bug in Bloop's argument parsing causes the
+  #    --nailgun-port argument to be lost in this case. Workaround this by prefixing
+  #    every main-destined argument with "--args"
+  define run_scala_main
+  	cd $(base_dir) && bloop --nailgun-port $(BLOOP_NAILGUN_PORT) run $(shell echo $(1) | sed 's/{.*}//') --main $(2)  $(addprefix --args ,$3)
+  endef
+  # recipes for generator and gg using bloop fall through to run_scala_main using Bloop
+  run_main = $(call run_scala_main,$(1),$(2),$(3))
+  PRE_ELABORATION_TARGETS += $(SCALA_BUILDTOOL_DEPS)
```

Comment:
@davidbiancolin as I'm starting to use this more and push on the rules a bit, I see a problem with this code.

The `PRE_ELABORATION_TARGETS` are used in the dependence list of target(s) in the TARGET_PROJECT_MAKEFRAG.  Dependence lists of targets are expanded when the target is defined in the makefile (assuming we aren't using .SECOND_EXPANSION) and thus this assignment of PRE_ELABORATION_TARGETS needs to happen before the `include $(TARGET_PROJECT_MAKEFRAG)`.

So does the assigment down on the new line 77.